main.tf

```diff
@@ -88,15 +92,18 @@ module "distribution_label" {
   tags       = "${var.tags}"
 }
 
-resource "null_resource" "default" {
-  triggers {
-    bucket             = "${element(compact(concat(list(var.origin_bucket), aws_s3_bucket.origin.*.bucket)), 0)}"
-    bucket_domain_name = "${format(var.bucket_domain_format, element(compact(concat(list(var.origin_bucket), aws_s3_bucket.origin.*.bucket)), 0))}"
-  }
+# ec2-downloads-windows is a bucket owned by amazon that will perminantly exist
+# It allows for the data source to be called during the destruction process without failing.
+# It doesn't get used for anything else.
+data "aws_s3_bucket" "selected" {
+  bucket = "${local.bucket == "" ? "ec2-downloads-windows" : local.bucket}"
```

Comment:
What if we created a new variable and called it `permanent_s3_bucket` that had a `description` of what you have above. Then we can reference `permanent_s3_bucket` everywhere we need the hack. I think this would make the hack very clear.

Reply:
Maybe use the `aws-cli` bucket? That would maybe raise less eyebrows. https://docs.aws.amazon.com/cli/latest/userguide/awscli-install-bundle.html